addons/point_of_sale/models/pos_session.py

```diff
@@ -1031,36 +1054,37 @@ def _accumulate_amounts(self, data):
                         combine_receivables_pay_later[payment_method] = self._update_amounts(combine_receivables_pay_later[payment_method], {'amount': amount}, date)
 
             if not order_is_invoiced:
-                order_taxes = defaultdict(tax_amounts)
-                for order_line in order.lines:
-                    line = self._prepare_line(order_line)
+                base_lines = order._prepare_tax_base_line_values(sign=1)
+                tax_results = self.env['account.tax']._compute_taxes(base_lines, order.company_id)
+                for base_line, to_update in tax_results['base_lines_to_update']:
                     # Combine sales/refund lines
                     sale_key = (
                         # account
-                        line['income_account_id'],
+                        base_line['account'].id,
                         # sign
-                        -1 if line['amount'] < 0 else 1,
+                        -1 if base_line['is_refund'] else 1,
                         # for taxes
-                        tuple((tax['id'], tax['account_id'], tax['tax_repartition_line_id']) for tax in line['taxes']),
-                        line['base_tags'],
+                        tuple(base_line['record'].tax_ids_after_fiscal_position.flatten_taxes_hierarchy().ids),
+                        tuple(to_update['tax_tag_ids'][0][2]),
+                    )
+                    sales[sale_key] = self._update_amounts(
+                        sales[sale_key],
+                        {'amount': to_update['price_subtotal']},
+                        order.date_order,
+                    )
+
+                # Combine tax lines
+                for tax_line in tax_results['tax_lines_to_add']:
+                    tax_key = (
+                        tax_line['account_id'],
+                        tax_line['tax_repartition_line_id'],
+                        tuple(tax_line['tax_tag_ids'][0][2]),
```

Comment:
mmh don't we need tax['id'] in tax_key ? it was there before :thinking:

Reply:
@qdp-odoo I'm fixing a big problem for the closing entry here.
Before:
The amounts was aggregated by taxes first, then spread into repartition lines.
Now:
The amounts are directly splited and rounded for each repartition lines in a way that the sum of taxes for each orders give the reported taxes in the closing entry.

Indeed, before, that wasn't true meaning if you had a closing entry in which you ask an invoice for all orders, the tax amounts wasn't correctly reset to 0 on the closing entry and then, you were creating ghost tax amounts.